exercises/react/src/example/kotlin/React.kt

```diff
@@ -0,0 +1,72 @@
+class Reactor<T>() {
+    abstract inner class Cell {
+        abstract val value: T
+        internal val dependents = mutableListOf<ComputeCell>()
+        internal fun eachDependent(f: (ComputeCell) -> Any) {
+            for (dep in dependents) {
+                f(dep)
+            }
+        }
+    }
+
+    interface Subscription {
+        fun cancel()
+    }
+
+    inner class InputCell(initialValue: T) : Cell() {
+        override var value: T = initialValue
+            set(newValue) {
+                field = newValue
+                eachDependent { it.propagate() }
+                eachDependent { it.fireCallbacks() }
+            }
+    }
+
+    inner class ComputeCell private constructor(val newValue: () -> T) : Cell() {
+        override var value: T = newValue()
+            private set
+
+        private var lastCallbackValue = value
+        private var callbacksIssued = 0
+        private val activeCallbacks = mutableMapOf<Int, (T) -> Any>()
+
+        constructor(c: Cell, f: (T) -> T) : this({ f(c.value) }) {
+            c.dependents.add(this)
+        }
+        constructor(c1: Cell, c2: Cell, f: (T, T) -> T) : this({ f(c1.value, c2.value) }) {
+            c1.dependents.add(this)
+            c2.dependents.add(this)
+        }
+
+        fun addCallback(f: (T) -> Any): Subscription {
```

Comment:
> Do you think we should provide some initial getting started help by at least providing some interfaces?

I would be in support of that, because it may be otherwise difficult to determine from just the tests that addCallback is supposed to return something that you can call `cancel()` on. It's easy enough to read function signatures if the expected return value is just a primitive, but if it needs to conform to an interface, then help is good.

By the way, it's time for me to point out the master list of all tracks' discussions on providing stubs https://github.com/exercism/discussions/issues/114

I'll let y'all decide what y'all want to do with the other exercises, but for this one, how far should we go?

* only provide the Subscription interface, and specify that `addCallback` should return something that conforms to it?
* provide Subscription, and stubs for `Cell`, `ComputeCell`, `InputCell`? But be careful, because this could constrain the student's implementation; maybe they don't want to have those exact three classes (maybe they just want to have a single `Cell` class?

I'll start with the first option and see how it goes.